src/attestation_types/quote/mod.rs

```diff
@@ -79,3 +124,89 @@ pub struct Quote {
+
+        let expected_quote_len = QUOTE_SIG_START + sd_len as usize;
+
+        if bytes.len() < expected_quote_len {
+            return Err(QuoteError(
+                "Insufficient Quote length; cannot convert from byte slice".to_string(),
+            ));
+        }
+
+        // Convert the Quote from byte slice
+        let mut header = [0u8; QUOTE_HEADER_LEN];
+        header.copy_from_slice(&bytes[0..QUOTE_HEADER_LEN]);
+
+        let mut body = [0u8; ISV_ENCLAVE_REPORT_LEN];
+        body.copy_from_slice(&bytes[QUOTE_HEADER_LEN..(QUOTE_HEADER_LEN + ISV_ENCLAVE_REPORT_LEN)]);
+
+        Ok(Self {
+            header: QuoteHeader::try_from(&header)?,
+            isv_enclave_report: Body::try_from(&body)?,
+            sig_data_len: SigDataLen::from(sd_len),
+            sig_data: SigData::try_from(&bytes[QUOTE_SIG_START..expected_quote_len])?,
+        })
+    }
+}
+
+impl Quote {
+    /// This vector of the Quote Header and ISV Enclave Report is the material signed
+    /// by the Quoting Enclave's Attestation Key and should be returned in raw form to
+    /// verify the Attestation Key's signature. Specifically, the header's version
+    /// number should also be kept intact in the vector, rather than being abstracted
+    /// into the Header enum.
+    pub fn raw_header_and_body(quote: &[u8]) -> Result<&[u8], QuoteError> {
+        if quote.len() < QUOTE_HEADER_LEN + ISV_ENCLAVE_REPORT_LEN {
+            return Err(QuoteError(
+                "Insufficient Quote length; cannot return raw header and body".to_string(),
+            ));
+        }
+
+        Ok(&quote[0..QUOTE_HEADER_LEN + ISV_ENCLAVE_REPORT_LEN])
+    }
+
+    /// The Report Data of the QE Report holds a SHA256 hash of (ECDSA Attestation Key || QE
+    /// Authentication data) || 32-0x00's. This hash must be verified for attestation.
+    /// The Report comes after the ISV Enclave Report Signature and Attestation Public Key in the
+    /// Quote Signature. The structure of the QE Report in the Quote Signature is identical
+    /// to the structure of any enclave's Report, so the Report Data begins at byte 320 of the Report.
+    pub fn raw_pck_hash(quote: &[u8]) -> Result<&[u8], QuoteError> {
+        if quote.len() < QE_REPORTDATA_START + PCK_HASH_LEN {
+            return Err(QuoteError(
+                "Insufficient Quote length; cannot return raw PCK hash".to_string(),
+            ));
+        }
+
+        Ok(&quote[QE_REPORTDATA_START..QE_REPORTDATA_START + PCK_HASH_LEN])
+    }
+
+    /// Retrieves Quote Header
+    pub fn get_header(&self) -> QuoteHeader {
+        self.header
+    }
+
+    /// Retrieves Quote Body
+    pub fn get_body(&self) -> Body {
+        self.isv_enclave_report
+    }
+
+    /// Retrieves Quote's sig length
+    pub fn get_siglen(&self) -> SigDataLen {
+        self.sig_data_len
+    }
+
+    /// Retrieves Quote's signature data
+    pub fn get_sigdata(&self) -> &SigData {
+        &self.sig_data
+    }
```

Comment:
I think the idiom for getters in Rust is to not use a `get_` prefix, so `get_body` -> `body`.

Should `get_header`, `get_body`, and `get_siglen` return a borrow? `get_sigdata` does. If these components are `Copy`, not returning a borrow could result in a potentially expensive copy.
```suggestion
    pub fn header(&self) -> &QuoteHeader {
        self.header
    }

    /// Retrieves Quote Body
    pub fn body(&self) -> &Body {
        self.isv_enclave_report
    }

    /// Retrieves Quote's sig length
    pub fn siglen(&self) -> &SigDataLen {
        self.sig_data_len
    }

    /// Retrieves Quote's signature data
    pub fn sigdata(&self) -> &SigData {
        &self.sig_data
    }
```